### PR TITLE
Replace `actions-rs/toolchain` with `dtolnay/rust-toolchain`

### LINKED ISSUE #1249: Replace `actions-rs/toolchain` with `dtolnay/rust-toolchain`.
The [acrtions-rs/toolchain](https://github.com/actions-rs/toolchain/issues/216) project is not currently maintained.
Consider changing it to [dtolnay/rust-toolchain](https://github.com/dtolnay/rust-toolchain).

### PR DESCRIPTION
**Pull request by jongwooo**
## Description

Resolve #1249 

The [actions-rs/toolchain](https://github.com/actions-rs/toolchain/issues/216) project is not currently maintained, so I replaced `actions-rs/toolchain` with `dtolnay/rust-toolchain`.

**AS-IS**

```yaml
- uses: actions-rs/toolchain@v1
  with:
    toolchain: stable
```

> `actions-rs/toolchain` can set up a profile. (e.g. default, minimal)

```yaml
- uses: actions-rs/toolchain@v1.0.6
  with:
    profile: minimal
    default: true
    components: clippy
```

**TO-BE**

```yaml
- uses: dtolnay/rust-toolchain@stable
```

> `dtolnay/rust-toolchain` ONLY install "minimal"

```yaml
- uses: dtolnay/rust-toolchain@master
  with:
    toolchain: 1.68
    components: clippy
```

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -20,10 +20,8 @@ jobs:
     steps:
       - uses: actions/checkout@v3
       - name: Add clippy
-        uses: actions-rs/toolchain@v1.0.6
+        uses: dtolnay/rust-toolchain@master
         with:
-          profile: minimal
-          default: true
```
Comment (jongwooo) [inline .github/workflows/rust.yml:26]: The `dtolnay/rust toolchain` does not have these inputs.

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -20,10 +20,9 @@ jobs:
...
+          toolchain: 1.68
```
Comment (jongwooo) [inline .github/workflows/rust.yml:25]: The `actions-rs/toolchain` read version from `rust-toolchain` if no toolchain input was specified. However, `dtolnay/rust-toolchain`does not have this feature. And the toolchain version for `dtolnay/rust-toolchain@1.68` is set to 1.68.2, not 1.68.

Comment (jongwooo) [reply .github/workflows/rust.yml:25]: @ever0de The output `name` of `dtolnay/rust-toolchain` is the same as the input `toolchain`. How about reading the value of `rust-toolchain` and putting it as input like below?

```yaml
- name: Read rust-toolchain file
  id: rust-toolchain
  run: echo "toolchain=$(cat rust-toolchain)" >> $GITHUB_OUTPUT
- name: Add clippy
  uses: dtolnay/rust-toolchain@master
  with:
    toolchain: ${{ steps.rust-toolchain.outputs.toolchain }}
    components: clippy
```

Comment (ever0de) [reply .github/workflows/rust.yml:25]: Oh yeah, that's good, I'm in favor of that as well.  Can you just test that it reads fine in CI?

Comment (jongwooo) [reply .github/workflows/rust.yml]: @ever0de Sure :)

**Review by ever0de — APPROVED** [review]
I think it's good :+1: Thanks for the contribution!

File: .github/workflows/coverage.yml [hunk 3]
```diff
@@ -18,9 +18,12 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
-      - uses: actions-rs/toolchain@v1
+      - name: Read rust-toolchain file
+        id: rust-toolchain
+        run: echo "toolchain=$(cat rust-toolchain)" >> $GITHUB_OUTPUT
```
Comment (jongwooo) [inline .github/workflows/coverage.yml:23]: Currently, the workflow reads the `rust-toolchain` file directly for usage. This restricts its scalability for changes like using the `rust-toolchain.toml` file instead. I'll find a better solution and let you know (maybe create a new action).

**Review by panarch — APPROVED** [review]
Looks all nice! 
Thanks for the contribution 👍 👍

**Pull request merged**